src/core/utils.ts

```diff
@@ -28,3 +29,13 @@ export function mapToSearchActions(values: string[], actions: typeof FluxCapacit
 }
 
 export const rayify = <T>(values: T | T[]): T[] => Array.isArray(values) ? values : [values];
+
+export const arrayToDotNotation = (input: string, throwError = false): string => {
+    if (typeof input !== 'string'){
+      if (throwError) {
+        throw (Error('Error input not a string'));
+      } else return <any>-1;
+    }
+
+    return input.replace(/\[([^\[\]]+)\]/g,'.$1').replace(/^[.]*|[.]*$/g,'');
```

Comment:
this RegExp should be saved in a constant at the top of this file so it doesn't need to be re-calculated

Reply:
what is the second replace here meant to be for?

Reply:
also, this functionality is just for arrays so we don't need to accept string keys within the array brackets, so I would suggest using the more succinct RegExp i suggested originally

Reply:
The second regex was supposed to strip leading and trailing zeroes so cases like '[a][b]' would be allowed, but I guess that is not needed so I will remove it.

Reply:
I've kept the first replace regex to allow strings within the brackets for more flexibility, is that fine or should it be changed to strictly allow digits?